sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -331,7 +319,44 @@ class HDFSFileCatalog(
     }
   }
 
-  protected def prunePartitions(
+  override def allFiles(): Seq[FileStatus] = leafFiles.values.toSeq
+
+  override def getStatus(path: Path): Array[FileStatus] = leafDirToChildrenFiles(path)
+
+  protected def inferPartitioning(): PartitionSpec = {
```

Comment:
@marmbrus These methods are basically unchanged, just moved around from HDFSFileCatalog to PartitioningAwareFileCatalog.